src/workloads/execution/PipelineUpdate.yml

```diff
@@ -0,0 +1,102 @@
+    DocumentCount: 10000
+    BatchSize: 2000
+    Document: &Document
+      t: &partitionID {^RandomInt: {min: 0, max: 100}}
+      randomID: &randomID {^RandomInt: {min: 0, max: 10000}}
+      int: &integer {^RandomInt: {min: 0, max: 100}}
+      str: &string {^RandomString: {length: {^RandomInt: {min: 20, max: 50}}}}
+      largeStr: &largeString {^RandomString: {length: 1024}}
+      obj: &object {a: *integer, b: *integer, c: *string}
+      arr: &array [*integer, *string, *object, *largeString]
+      nestedObj: &nestedObject {
+        int: *integer,
+        str: *string,
+        obj: *object,
+        arr: [1, 2, 3],
+        nestedArr: [*integer, *string, *object, *array],
+        largeStr: *largeString}
+    Indexes:
+    - keys: {t: 1}
+  - &Nop {Nop: true}
+  - *Nop
+
+- Name: RunUpdates
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1000    # Classic updates phase.
+    Database: *db
+    Operations:
+    - OperationMetricsName: ClassicUpdateAddFields
+      OperationName: RunCommand
+      OperationCommand:
+        update: Collection0
+        updates: [{q: {t: *partitionID}, u:
+                     {$set: {int: *integer, string: *string}}, multi: true}]
+        writeConcern: {w: majority}
+    - OperationMetricsName: ClassicUpdateModifyArray
+      OperationName: RunCommand
+      OperationCommand:
+        update: Collection0
+        updates: [{q: {t: *partitionID}, u:
+                     {$set: {nestedObj.arr: [*integer, *integer, *integer]}}, multi: true}]
+        writeConcern: {w: majority}
+    - OperationMetricsName: ClassicUpdateUnset
+      OperationName: RunCommand
+      OperationCommand:
+        update: Collection0
+        updates: [{q: {t: *partitionID}, u: {$unset: {string: "", int: ""}}, multi: true}]
+        writeConcern: {w: majority}
+    - OperationMetricsName: ClassicUpdateReplaceDocument
+      OperationName: RunCommand
+      OperationCommand:
+        update: Collection0
+        updates: [{q: {randomID: *randomID}, u: *Document}]
+        writeConcern: {w: majority}
+  - Repeat: 1000    # Pipeline-based updates phase.
+    Database: *db
+    Operations:
+    - OperationMetricsName: PipelineUpdateAddFields
+      OperationName: RunCommand
+      OperationCommand:
+        update: Collection0
+        updates: [{q: {t: *partitionID}, u: [
+          {$set: {int: *integer, string: *string}}], multi: true}]
+        writeConcern: {w: majority}
+    - OperationMetricsName: PipelineUpdateModifyArray
+      OperationName: RunCommand
+      OperationCommand:
+        update: Collection0
+        updates: [{q: {t: *partitionID}, u: [
+          {$set: {nestedObj.arr: [*integer, *integer, *integer]}}], multi: true}]
+        writeConcern: {w: majority}
+    - OperationMetricsName: PipelineUpdateUnset
+      OperationName: RunCommand
+      OperationCommand:
+        update: Collection0
+        updates: [{q: {t: *partitionID}, u: [{$unset: [string, int]}], multi: true}]
```

Comment:
[question] are you sure this one works? I would have thought 'string' and 'int' would need quotes around them. A lot of this stuff is magic to me though so I could be wrong.

Reply:
yea it will quote "string" and "int" when generating commands.

You can also check the log on [Evergreen](https://evergreen.mongodb.com/task_log_raw/sys_perf_linux_3_node_replSet_pipeline_update_patch_4a95b7ebc88499c2b3b9aaa11fabbed895f40f18_5f4516fce3c33163f25b4a75_20_08_25_13_51_29/2?type=T) that gives the generated commands: 

> [2020/08/25 20:51:12.208] 20:51:12Z>  [2020-08-25 20:51:12.207972] [0x00007f37061ee700] [info]     Running command: { "update" : "Collection0", "updates" : [ { "q" : { "t" : 64 }, "u" : [ { "$unset" : [ "string", "int" ] } ], "multi" : true } ], "writeConcern" : { "w" : "majority" } } on database: test

Reply:
Ah, nice. Thanks for looking into it.